### PR TITLE
[Item] Add new indicator to Item

### PR DESCRIPTION
**Pull request by dev-nat-ro**
### WHY are these changes introduced?

Part of https://github.com/Shopify/web/issues/84806

We need a way to let our merchants know there are new filters available for them to use.

### WHAT is this pull request doing?

- Adding the indicator to the AlphaFilter component, the respective test and story.
- Adding the Suffix to the FilterInterface to allow AlfaTabs to display one.

Screenshot 2023-05-09 at 4 01 36 PM

### How to 🎩

Check the [storybook story](https://polaris-qyzt.natalia-ramirez.us.spin.dev/) added: AlphaFilters -> With an indicator

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by AndrewMusgrave — APPROVED** [review]
Nice work 🙌

**Comment by mattkubej** [comment]
Is there a spin instance running storybook available that I can use to vet the change?

File: polaris-react/src/types.ts [hunk 1]
```diff
@@ -398,6 +398,8 @@ export interface FilterInterface {
    * Whether or not the clear button is displayed
    */
   hideClearButton?: boolean;
+  /** Suffix source */
+  suffix?: React.ReactNode;
```
Comment (mattkubej) [inline polaris-react/src/types.ts:402]: Why was this attribute added to the interface? It doesn't seem like it's being leveraged elsewhere within this PR

Comment (dev-nat-ro) [reply polaris-react/src/types.ts:402]: Adding the suffix type allows AlfaTabs to receive that prop from web, which will allow me to implement the following UI, in a separate PR:

Screenshot 2023-05-11 at 11 18 31 AM

**Comment by dev-nat-ro** [comment]
@mattkubej  Thank you for your review, I've added a link in the description and answered your question about the suffix.
To be transparent, this PR inspired a conversation with Polaris and QC which you can follow in slack at `#temp-new-feature-pips` which might result in changes or not merging it at all.

File: polaris-react/src/types.ts (hunk 1 above)
Comment (mattkubej) [reply polaris-react/src/types.ts:402]: So, are you adding the property to the interface now in anticipation of following up with supporting implementation in a subsequent pull request to Polaris? If that is the case, why not add this property along with the implementation, then utilize within web when it's usable?

Alternatively, if you're adding this property to the interface and depending on the interface within web for a custom implementation of a Filter, then why not extend the interface and add your property to your custom implementation?

**Comment by mattkubej** [comment]
> I've added a link in the description

Thanks @nat-king! What subdomain is storybook available on within that spin instance? I'm not seeing a link to storybook on the landing page and the link to the polaris landing page is pointing to an instance that is not running.

**Comment by dev-nat-ro** [comment]
> > I've added a link in the description
> 
> Thanks @nat-king! What subdomain is storybook available on within that spin instance? I'm not seeing a link to storybook on the landing page and the link to the polaris landing page is pointing to an instance that is not running.

My spin instance broke, I'm working on fixing it right now.

**Comment by dev-nat-ro** [comment]
Closing this PR given the response provided in slack:

>TLDR – We are in alignment that this should not receive any new feature awareness such as a badge or a pip.
Rationale, thoughts and general guidance
Merchants who need this will already be going to this feature and should discover the new filter(s) organically.
The added visual noise outweighs the value/impact of the changes. The impact to noise ratio should be extremely high.
As Shopify positions itself to be more of a pro tool, we need to build, ship and communicate in that light. In feature rich pro tools like Figma, Adobe products, gmail or iOS only the biggest and most important changes to the product get highlighted and even this is mostly outside of the product itself in marketing, presentations and on websites. The vast majority of features just ship and are ready to be discovered. We should do the same.
Polaris will be publishing [new guidance](https://github.com/Shopify/polaris/pull/9186) to our site shortly that gives this same framework to everyone.
We do believe there may be value in some sort of admin wide change log or systematic way to share smaller individual feature enhancements but right now that doesn’t exist and there aren’t plans that we are aware of to build such a feature.
The bar to highlight something new inside the product is high and is getting higher as more teams are reaching for this pattern.
There is no doubt that this work will be impactful and you should all be excited to get this work in the hands of our merchants.

Thanks for the reviews anyway @AndrewMusgrave and @mattkubej
